packages/design-system/src/scalars/components/amount-field/use-amount-field.ts

```diff
@@ -1,40 +1,137 @@
 
   const valueInput =
     type === "Amount" || type === "AmountPercentage"
       ? (currentValue as number)
-      : (currentValue as AmountCurrency).amount;
+      : (currentValue as AmountCurrency | AmountToken).amount;
+
   const isPercent = type === "AmountPercentage";
-  const isCurrency = type === "AmountCurrency";
 
-  const options = getLabelValueCurrenct(allowedCurrencies);
-  const currency =
-    type === "AmountCurrency" ? (currentValue as AmountCurrency).currency : "";
+  //Allow select only if type is AmountCurrency or AmountToken
+  const isShowSelect = type === "AmountCurrency" || type === "AmountToken";
+
+  // Filter only if allowedCurrencies is provided
+  const optionsCurrencies = getCountryCurrencies(allowedCurrencies);
+  const optionsTokens = getTokens(allowedTokens);
+
+  const options = type === "AmountCurrency" ? optionsCurrencies : optionsTokens;
+
+  const valueSelect =
+    type === "AmountCurrency"
+      ? (currentValue as AmountCurrency).currency
+      : (currentValue as AmountToken).token;
+
+  // Handle the change of the input
+  const handleOnChangeInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    let newValue: AmountValue = {} as unknown as AmountValue;
+    if (type === "AmountCurrency" && typeof value === "object") {
+      newValue = {
+        ...value,
+        amount: Number(e.target.value),
+      } as AmountValue;
+    }
+    if (type === "AmountToken" && typeof value === "object") {
+      newValue = {
+        ...value,
+        amount: BigInt(e.target.value),
+      } as AmountValue;
+    }
+    if (type === "Amount" || type === "AmountPercentage") {
+      newValue = Number(e.target.value);
+    }
+    //Create the event
+    const nativeEvent = new Event("change", {
+      bubbles: true,
+      cancelable: true,
+    });
+    Object.defineProperty(nativeEvent, "target", {
+      value: { value: newValue },
+      writable: false,
+    });
+    onChange?.(newValue as unknown as React.ChangeEvent<HTMLInputElement>);
+  };
 
-  const isSearchable = options.length >= 5;
+  // Handle the change of the select
+  const handleOnChangeSelect = (e: string | string[]) => {
+    let newValue: AmountCurrency | AmountToken = {} as
+      | AmountCurrency
+      | AmountToken;
+    if (type === "AmountCurrency" && typeof value === "object") {
+      newValue = {
+        ...value,
+        currency: typeof e === "string" ? e : undefined,
+      } as AmountCurrency;
+    }
+    if (type === "AmountToken" && typeof value === "object") {
+      newValue = {
+        ...value,
+        token: typeof e === "string" ? e : undefined,
+      } as AmountToken;
+    }
+
+    //Create the event
+    const nativeEvent = new Event("change", {
+      bubbles: true,
+      cancelable: true,
+    });
+    Object.defineProperty(nativeEvent, "target", {
+      value: { value: newValue },
+      writable: false,
+    });
+    onChange?.(newValue as unknown as React.ChangeEvent<HTMLInputElement>);
```

Comment:
similar comment that above, we're sending the value instead the event

Reply:
You're right—thank you for catching that. Done